solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -232,15 +230,20 @@ public void setConfigMetadata(String configName, Map<String, Object> data) throw
 
   @Override
   public Map<String, Object> getConfigMetadata(String configName) throws IOException {
+    byte[] data = null;
     try {
-      @SuppressWarnings("unchecked")
-      Map<String, Object> data =
-          (Map<String, Object>)
-              Utils.fromJSON(zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true));
-      return data;
+      data = zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true);
+    } catch (KeeperException.NoNodeException e) {
+      return new HashMap<>();
```

Comment:
I like we simplified the lines.   Should we return from the exception, or just catch it since below we check `data == null` and then return?

Reply:
previously NoNodeException is not checked.